com.unity.multiplayer.mlapi/Runtime/Core/RPCQueue/RPCQueueProcessing.cs

```diff
@@ -0,0 +1,246 @@
+using Unity.Profiling;
+using MLAPI.Configuration;
+using MLAPI.Messaging;
+using MLAPI.Profiling;
+using MLAPI.Serialization.Pooled;
+
+
+namespace MLAPI
+{
+    /// <summary>
+    /// RPCQueueProcessing
+    /// Handles processing of RPCQueues
+    /// Inbound to invocation
+    /// Outbound to send
+    /// </summary>
+    internal class RPCQueueProcessing
+    {
+
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+        static ProfilerMarker s_MLAPIRPCQueueProcess = new ProfilerMarker("MLAPIRPCQueueProcess");
+        static ProfilerMarker s_MLAPIRPCQueueSend = new ProfilerMarker("MLAPIRPCQueueSend");
+#endif
+
+        //NSS-TODO:  Need to determine how we want to handle all other MLAPI send types
+        //Temporary place to keep internal MLAPI messages
+        readonly List<FrameQueueItem> InternalMLAPISendQueue = new List<FrameQueueItem>();
+
+        RPCQueueManager rpcQueueManager;
+
+        /// <summary>
+        /// ProcessReceiveQueue
+        /// Public facing interface method to start processing all RPCs in the current inbound frame
+        /// </summary>
+        public void ProcessReceiveQueue()
+        {
+            RPCReceiveQueueProcessFlush();
+        }
+
+        /// <summary>
+        /// RCPQueueReeiveAndFlush
+        /// Parses through all incoming RPCs in the active RPC History Frame (RPCQueueManager)
+        /// </summary>
+        private void RPCReceiveQueueProcessFlush()
+        {
+            bool AdvanceFrameHistory = false;
+            RPCQueueManager rpcQueueManager = NetworkingManager.Singleton.GetRPCQueueManager();
+            if(rpcQueueManager != null)
+            {
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+            s_MLAPIRPCQueueProcess.Begin();
+#endif
+                try
+                {
+
+                    QueueHistoryFrame CurrentFrame = rpcQueueManager.GetCurrentFrame(QueueHistoryFrame.QueueFrameType.Inbound);
+                    if(CurrentFrame != null)
+                    {
+                        FrameQueueItem currentQueueItem = CurrentFrame.GetFirstQueueItem();
+                        while(currentQueueItem.QueueItemType != RPCQueueManager.QueueItemType.NONE)
+                        {
+                            AdvanceFrameHistory = true;
+                            if(rpcQueueManager.IsLoopBack())
+                            {
+                                currentQueueItem.ItemStream.Position = 1;
+                            }
+
+                            NetworkingManager.Singleton.InvokeRPC(currentQueueItem);
+                            ProfilerStatManager.rpcsQueueProc.Record();
+                            currentQueueItem = CurrentFrame.GetNextQueueItem();
+                        }
+                        //We call this to dispose of the shared stream writer and stream
+                        CurrentFrame.CloseQueue();
+                    }
+
+                }
+                catch(Exception ex)
+                {
+                    Debug.LogError(ex);
```

Comment:
Should we crash?  Is it desirable to silently absorb the exception?

Reply:
Actually, this "should never crash" but since new systems and such are being built around it I had left the try catch in for the time being.  It is not "best practice", but yes I do think if we want to wrap the try catch in #if DEVELOPMENT_BUILD || UNITY_EDITOR in order to see any crashes happening during this time but just crash in a release build that would most likely be appropriate.  
(or we just remove that try catch before it hits master)

Reply:
Yeah, I'd totally support removing the `try` so it can fail loud and proud somewhere else